app/ts/utils/augurContractUtils.ts

```diff
@@ -12,86 +12,76 @@ import { REPORTING_PARTICIPANT_ABI } from '../ABI/ReportingParticipant.js'
 import { REPUTATION_TOKEN_ABI } from '../ABI/ReputationToken.js'
 import { REDEEM_STAKE_ABI } from '../ABI/RedeemStakeAbi.js'
 import { AUDIT_FUNDS_ABI } from '../ABI/AuditFunds.js'
-import { createReadClient, createWriteClient } from './ethereumWallet.js'
+import { ReadClient, WriteClient } from './ethereumWallet.js'
 import { UNIVERSE_ABI } from '../ABI/Universe.js'
 import { getAllPayoutNumeratorCombinations } from './augurUtils.js'
 import { encodePacked, keccak256 } from 'viem'
 
-export const createYesNoMarket = async (universe: AccountAddress, marketCreator: AccountAddress, endTime: bigint, feePerCashInAttoCash: bigint, affiliateValidator: AccountAddress, affiliateFeeDivisor: bigint, designatedReporterAddress: AccountAddress, extraInfo: string) => {
-	const client = createWriteClient(marketCreator)
-	const { request } = await client.simulateContract({
-		account: marketCreator,
+export const createYesNoMarket = async (universe: AccountAddress, writeClient: WriteClient, endTime: bigint, feePerCashInAttoCash: bigint, affiliateValidator: AccountAddress, affiliateFeeDivisor: bigint, designatedReporterAddress: AccountAddress, extraInfo: string) => {
+	await writeClient.writeContract({
 		address: universe,
 		abi: AUGUR_UNIVERSE_ABI,
 		functionName: 'createYesNoMarket',
 		args: [endTime, feePerCashInAttoCash, affiliateValidator, affiliateFeeDivisor, designatedReporterAddress, extraInfo]
 	})
-	await client.writeContract(request)
 }
```

Comment:
I thought the point of this change was to make it so we don't parse the ABI every function call, but it appears we are still parsing the ABI every function call here.

My hope is that this code can be changed to something like, or just remove the need for this function entirely:
```ts
export const createYesNoMarket = async (universe: AccountAddress, writeClient: WriteClient, endTime: bigint, feePerCashInAttoCash: bigint, affiliateValidator: AccountAddress, affiliateFeeDivisor: bigint, designatedReporterAddress: AccountAddress, extraInfo: string) => {
	await getMemoizedUniverseContract().createYesOrNoMarket(endTime, feePerCashInAttoCash, affiliateValidator, affiliateFeeDivisor, designatedReporterAddress, extraInfo)
}
```

Maybe I am misunderstanding this PR's purpose?

Reply:
this pr doesn't recreate the write/read clients all the time, but still does the abi stuff. The ABI parsing issue is this: https://github.com/AugurProject/Augur-Constant-Product-Market/issues/52 I don't have a solution for it atm, it needs more research and maybe changes in viem(?)